.github/workflows/test.yml

```
@@ -1,25 +1,45 @@
-name: test
+name: Test Matrix
+
 on:
-- pull_request
+  pull_request:
+  push: { branches: [ master ] }
+    
 jobs:
-  vapor_macos:
-    runs-on: macos-latest
-    env:
-      DEVELOPER_DIR: /Applications/Xcode_11.4.1.app/Contents/Developer
-    steps:
-    - uses: actions/checkout@v2
-    - run: xcrun swift test --enable-test-discovery --sanitize=thread
-  vapor_xenial:
-    container: 
-      image: vapor/swift:5.2-xenial
+
+  All-tests-linux:
```

What about just `linux` / `macOS` for the names?

---

Works for me. I'm still hoping to discover a way to add the macOS job to the Linux matrix and get rid of all the pointless repetition. I mean okay, GH workflows don't support the wacko "merge key" `<< :` YAML extension, but they could at _LEAST_ support basic `&` anchors and `*` alias nodes...

```suggestion
linux:
```